### PR TITLE
Images feature

### LINKED ISSUE #50: Upload images
As an interior designer, I want to upload images for my contact, so that I can reference images that are relevant to the contact such as floorplan, design inspirations or colour schemes.

### PR DESCRIPTION
**Pull request by takufunkai**
Fixes #50 

Features added for this PR
- [x] Add images
- [x] List images 
- [x] Delete images

image

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 1]
```diff
@@ -11,39 +11,45 @@
...
     private final String feedbackToUser;
 
-    /** Help information should be shown to the user. */
-    private final boolean showHelp;
+    public enum SpecialCommandResult {
+        SHOW_HELP, // Help information should be shown to the user
+        EXIT, // The application should exit
+        VIEW_IMAGES, // Should load up the contact's images
+        NONE
+    }
```
Comment (xSaints19x) [inline src/main/java/seedu/address/logic/commands/CommandResult.java:19]: Good use of enum to refactor code! 👍

File: src/main/java/seedu/address/model/image/ImageDetailsList.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * Returns an iterator over elements of type {@code Images}.
+     *
+     * @return an Iterator.
+     */
+    @Override
+    public Iterator<ImageDetails> iterator() {
+        return this.images.iterator();
+    }
+}
```
Comment (xSaints19x) [inline src/main/java/seedu/address/model/image/ImageDetailsList.java:84]: Good use of the iterator class here!

**Review by xSaints19x — APPROVED** [review]
Lgtm, awesome work! 💯

**Review by weijuey — CHANGES_REQUESTED** [review]
Great work Ezekiel, this was a pretty tough feature and I like that while you implemented the feature, you improved the current codebase for not just supporting the feature but also future extensions. Code quality was top notch as well :100:. Just a few nits and a query, otherwise, lgtm! :tada:

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 3]
```diff
@@ -11,39 +11,45 @@
 
     private final String feedbackToUser;
 
-    /** Help information should be shown to the user. */
-    private final boolean showHelp;
+    public enum SpecialCommandResult {
```
[the remark this answers is not in the document]
Comment (takufunkai) [reply src/main/java/seedu/address/logic/commands/CommandResult.java:14]: Thanks for the words of encouragement!

File: src/test/java/seedu/address/model/image/ImageDetailsTest.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    @Test
+    void getImageFile() {
+        String fileName = "name.test";
+        File file = Path.of("test", fileName).toFile();
+        ImageDetails imageDetails = new ImageDetails(file);
+        assertEquals(file, imageDetails.getImageFile());
+    }
+
+    @Test
+    void getAbsolutePath() {
```
[the remark this answers is not in the document]
Comment (takufunkai) [reply src/test/java/seedu/address/model/image/ImageDetailsTest.java:36]: Oops! Forgot to change after changing the method name. Thanks for the catch!

**Review by weijuey — APPROVED** [review]

**Pull request merged**